Lib/pydoc.py

```diff
@@ -2059,10 +2059,11 @@ def interact(self):
         while True:
             try:
                 request = self.getline('help> ')
-                if not request: break
             except (KeyboardInterrupt, EOFError):
                 break
             request = request.strip()
+            if not request:
+                continue
```

Comment:
We can potentially avoid creating a new string object:
```suggestion
            if not request or request.isspace():
                continue  # back to the prompt
            request = request.strip()
```

Reply:
Is avoiding creating that string object in the rare case where we're given an empty input really worth the extra code here?  The string object is going to be tiny in that case anyway, and it's unlikely for it to happen very often...